cmd/internal/register/maps.go

```diff
@@ -113,16 +131,47 @@ func Maps(a *atlas.Atlas, maps []config.Map, providers map[string]provider.Tiler
 
 			// confirm our providerLayer name is registered
 			var found bool
+			// this must be an array because auto providers may have multiple layers that match regex
+			var provLayers []string
 			var layerGeomType tegola.Geometry
-			for i := range layerInfos {
-				if layerInfos[i].Name() == providerLayer[1] {
-					found = true
+			for i, info := range layerInfos {
+				// check if provider layers are automatically generated
+				if auto {
+					// regex to check against. ^ to match only beginning of phrase
+					exp := fmt.Sprintf("^%v", providerLayer[1])
+					// if * (wildcard), select all
+					if exp == "^*" {
+						provLayers = append(provLayers, info.Name())
+						found = true
+						layerGeomType = info.GeomType()
+					} else {
+						// must compile the regex first before testing phrase
+						r, err := regexp.Compile(exp)
+						if err != nil {
+							log.Printf("Error when parsing regex (layer: %v): %v", info.Name(), err)
+						} else {
+							// if regex matches, push provider layer. Note that there is no break--we need to find all layers that match
+							if r.MatchString(info.Name()) {
+								provLayers = append(provLayers, info.Name())
+								found = true
+								// because this is the same for all auto provider layers, we don't need to worry about which loop we are on
+								layerGeomType = info.GeomType()
+							}
+						}
+					}
+
+				} else {
+					if layerInfos[i].Name() == providerLayer[1] {
+						found = true
+						provLayers = append(provLayers, providerLayer[1])
+						// read the layerGeomType
+						layerGeomType = layerInfos[i].GeomType()
+						break
 
-					// read the layerGeomType
-					layerGeomType = layerInfos[i].GeomType()
-					break
+					}
 				}
 			}
+
 			if !found {
 				return ErrProviderLayerNotRegistered{
```

Comment:
We should be able to get rid of the `found` variable and just check to see if `len(provLayers) > 0`.